### PR TITLE
feat(core): added new role

### PR DESCRIPTION
**Pull request by mattjoke**
* new role for basic auth accounts
* role skips all MFA checks

DEPLOYMENT NOTE: added new role 'PERUNADMINBA'

### CODE REVIEW

File: perun-base/src/main/resources/perun-roles.yml [hunk 1]
```diff
@@ -8181,13 +8186,33 @@ perun_roles_management:
       - PERUNADMIN:
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:8186]: Theoretically, PERUNADMINBA should be able to set PERUNADMIN, no?

File: perun-base/src/main/resources/perun-roles.yml [hunk 2]
```diff
@@ -8181,13 +8186,33 @@ perun_roles_management:
...
+      - PERUNADMINBA:
+    privileged_roles_to_read:
+      - PERUNADMIN:
+      - PERUNADMINBA:
+      - PERUNOBSERVER:
+    associated_read_roles:
+      - PERUNOBSERVER
+    assignable_to_attributes: false
+    mfa_critical_role: false
+    display_name: "Perun admin BA"
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:8214]:
```suggestion
    display_name: "Perun admin BA"
    skip_mfa: true
```
this is missing in order to skip MFA when performing critical actions

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java [hunk 3]
```diff
@@ -1289,7 +1289,7 @@ public static void unsetRole(PerunSession sess, User user, PerunBean complementa
 
 		try {
 			if(!authorizedToManageRole(sess, complementaryObject, role)) {
-				throw new PrivilegeException("You are not privileged to use the method unsetRole.");
+				throw new PrivilegeException("You are not privileged to use the method unsetRole. "+authorizedToManageRole(sess, complementaryObject, role));
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java:1292]: Why is this here

**Review by xflord — CHANGES_REQUESTED** [review]
Again, I would utilize the `isPerunAdmin()` method to reduce the explicit checks for PERUNADMINBA.

Also, add the changes to the perun-roles template in the ansible repo https://github.com/CESNET/ansible-role-perun-docker-server

File: perun-base/src/main/resources/perun-roles.yml (hunk 1 above)
Comment (mattjoke) [reply perun-base/src/main/resources/perun-roles.yml:8186]: I think this is not what we want because PERUNADMINBA is supposed to be just an internal role, which will be added manually through the database.

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java (hunk 3 above)
Comment (mattjoke) [reply perun-core/src/main/java/cz/metacentrum/perun/core/api/AuthzResolver.java:1292]: Good catch, just a leftover from debugging

**Comment by mattjoke** [comment]
I replaced role checks with `isAdmin()`. Nevertheless, most of the occurrences have explicit checks (isAuthorized, hasRole), which cannot be replaced with `isAdmin()`

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java [hunk 4]
```diff
@@ -376,7 +376,7 @@ public User createUser(PerunSession sess, User user) throws PrivilegeException {
 		Utils.notNull(user, "user");
 
 		// Authorization
-		if(!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN)) {
+		if(!(AuthzResolver.isAuthorized(sess, Role.PERUNADMIN) || AuthzResolver.isAuthorized(sess, Role.PERUNADMINBA))) {
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java:379]: No need for this - this form of isAuthorized is deprecated

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java [hunk 5]
```diff
@@ -579,7 +579,7 @@ public void sendMessage(PerunSession sess, Application app, MailType mailType, S
 		if (mail == null) throw new RegistrarException("Notification template for "+mailType+" is not defined.");
 		if (!mail.getSend()) throw new RegistrarException("Sending of notification "+mailType+" is disabled.");
 
-		if (!AuthzResolver.hasRole(sess.getPerunPrincipal(), Role.PERUNADMIN)) {
+		if (!(AuthzResolver.hasRole(sess.getPerunPrincipal(), Role.PERUNADMIN) || AuthzResolver.hasRole(sess.getPerunPrincipal(), Role.PERUNADMINBA))) {
```
Comment (xflord) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java:582]: This call effectively does the same as isPerunAdmin()

**Review by xflord — COMMENTED** [review]
I also noticed that the `isUserPerunAdmin()` exists, even though it is not used anywhere. Maybe we can mark it as deprecated to avoid misusage (since it only checks for PERUNADMIN). What do you think?

File: perun-base/src/main/resources/perun-roles.yml (hunk 1 above)
Comment (xflord) [reply perun-base/src/main/resources/perun-roles.yml:8186]: I know that, but as an internal "super user" account, I don't see why it shouldn't be allowed to set PERUNADMIN. But I guess it's not a big deal.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java (hunk 5 above)
Comment (mattjoke) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java:582]: Well, I think it is different because it does throw the error if the role does not exist in the Perun. `isPerunAdmin()` does not check if the role exists.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/SecurityTeamsManagerBlImpl.java [hunk 6]
```diff
@@ -52,7 +52,7 @@ public SecurityTeamsManagerBlImpl(SecurityTeamsManagerImplApi securityTeamsManag
 
 	@Override
 	public List<SecurityTeam> getSecurityTeams(PerunSession sess) {
-		if (AuthzResolverBlImpl.hasRole(sess.getPerunPrincipal(), Role.PERUNADMIN)) {
+		if (AuthzResolverBlImpl.hasRole(sess.getPerunPrincipal(), Role.PERUNADMIN) || AuthzResolverBlImpl.hasRole(sess.getPerunPrincipal(), Role.PERUNADMINBA)) {
```
[the remark this answers is not in the document]
Comment (mattjoke) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/SecurityTeamsManagerBlImpl.java:55]: Same here; the check throws an error, if the role is not in Perun.

**Review by bodnara — APPROVED** [review]
Should be ok like this.

**Review by sarkapalkovicova — APPROVED** [review]
I tested it and did not find any issues.
